test/integration/main_test.go

```diff
@@ -191,44 +191,141 @@ func duration(t *testing.T, start time.Time, end time.Time) float64 {
 }
 
 func LogChecker(t *testing.T) *LogCheck {
-	defaults := &LogCheck{
-		interval:   time.Second,
-		logOptions: corev1.PodLogOptions{},
-		timeout:    5 * time.Minute,
-		failFast:   true,
-		test:       t,
-		clientset:  clientset,
-	}
-	return defaults
+	return logChecker(t, clientset)
 }
 
 func checkPodsLogs(t *testing.T, message string) *LogCheck {
 	return LogChecker(t).Search(message)
 }
 
-func forceUpdateSecret(ns string, secretName string, secret *v1.Secret) error {
-	latestSecret, err := clientset.CoreV1().Secrets(ns).Get(context.Background(), secretName, metav1.GetOptions{})
+func tinyproxy(t *testing.T) *TinyProxy {
+	proxy := &TinyProxy{}
+	err := proxy.create(t, clientset)
+	e(t, err, "failed to create tinyproxy")
+	proxy.waitUntilReady()
+	return proxy
+}
+func (proxy *TinyProxy) setClusterWideProxy(t *testing.T) func() {
+	oldProxy, _ := configV1Client().Proxies().Get(context.Background(), "cluster", metav1.GetOptions{})
+	cwproxy := configv1.Proxy{Spec: configv1.ProxySpec{
+		HTTPProxy:          proxy.address,
+		HTTPSProxy:         proxy.address,
+		NoProxy:            "example.com",
+		ReadinessEndpoints: []string{"ohno"},
+		TrustedCA: configv1.ConfigMapNameReference{
+			Name: "service.default",
+		},
+	},
+	}
+	cwproxy.Name = "cluster"
+	//version, _ := strconv.Atoi(oldProxy.ResourceVersion)
+	//version++
+	cwproxy.ObjectMeta.ResourceVersion = oldProxy.ResourceVersion //string(rune(version))
+
+	_, err := configV1Client().Proxies().Update(context.Background(), &cwproxy, metav1.UpdateOptions{})
+	e(t, err, "tak ne XDDDDDDD")
```

Comment:
Please add some meaningful message here.

Reply:
Oh I see that I have totally missed some things, thanks for pointing out